### PR TITLE
Refactor flattenCollection to GeometryUtils

### PR DESCRIPTION
**Pull request by jagill**
We will need for this function in several places, and it is purely geometric.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by jagill** [comment]
cc: @aweisberg

File: presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/TestGeometryUtils.java [hunk 1]
```diff
@@ -69,4 +75,48 @@ private void assertGetExtent(String wkt, Rectangle expected)
...
+        assertFlattenLeavesUnchanged(OGCGeometry.fromText("POLYGON ((0 0, 0 1, 1 1, 0 0))"));
+        assertFlattenLeavesUnchanged(OGCGeometry.fromText("MULTIPOLYGON EMPTY"));
+        assertFlattenLeavesUnchanged(OGCGeometry.fromText("MULTIPOLYGON (((0 0, 0 1, 1 1, 0 0)))"));
+        assertFlattenLeavesUnchanged(OGCGeometry.fromText("MULTIPOLYGON (((0 0, 0 1, 1 1, 0 0)), ((10 10, 10 11, 11 11, 10 10)))"));
+
+        assertFlattens(OGCGeometry.fromText("GEOMETRYCOLLECTION EMPTY"), ImmutableList.of());
+        assertFlattens(OGCGeometry.fromText("GEOMETRYCOLLECTION (POINT EMPTY)"), OGCGeometry.fromText("POINT EMPTY"));
+        assertFlattens(OGCGeometry.fromText("GEOMETRYCOLLECTION (POINT (0 1))"), OGCGeometry.fromText("POINT (0 1)"));
+        assertFlattens(OGCGeometry.fromText("GEOMETRYCOLLECTION (GEOMETRYCOLLECTION EMPTY)"), ImmutableList.of());
+        assertFlattens(OGCGeometry.fromText("GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 1), POINT (1 2)))"),
```
Comment (highker) [inline presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/TestGeometryUtils.java:102]: put `OGCGeometry.fromText("GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 1), POINT (1 2)))"),` to its own line

**Comment by aweisberg** [comment]
Add me as a reviewer so I am less likely to miss it when I look for things awaiting review from em. @ing me is slight less reliable.

**Comment by jagill** [comment]
Test failure is this seemingly unrelated one:
```
[ERROR] testAggregationOverUnknown(com.facebook.presto.tests.TestNonIterativeDistributedQueries)  Time elapsed: 1.195 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT clerk, min(totalprice), max(totalprice), min(nullvalue), max(nullvalue) FROM (SELECT clerk, totalprice, null AS nullvalue FROM orders) GROUP BY clerk
```

**Pull request merged**